src/runtime/internal/dom.ts

```diff
@@ -1,5 +1,27 @@
 import { has_prop } from './utils';
 
+let appendStylesTo = document.head;
```

Comment:
I'm fairly confident this can't live in the global scope.  If I render one svelte component inside a shadow DOM, with the appropriate target, styles will continue to get appended to that context even if the next component gets rendered into light DOM.

Reply:
You are right. I assumed someone exports only a single svelte-component as an entry point. But if someone wants to have multiple entry-point components in a single file, the global variable would not work.
I think the only solution to this problem is then to pass a reference to the `target`-node down into each `create_fragment` call. I will update this PR.